### PR TITLE
[Torch] fixup onnx2trt int64 inputs

### PR DESCRIPTION
**Pull request by fortianyou**
+ `torch._C._jit_pass_lower_to_onnx` and `torch._C._jit_pass_onnx_scalar_type_analysis` APIs has changed in torch 1.9.0 https://github.com/alibaba/BladeDISC/issues/253
+ Cast subgraph's inputs to `torch.int32` where it's `torch.int64` https://github.com/alibaba/BladeDISC/issues/254

TorchBlade returns a model with accuracy error, after optimizing the hugging face a pre-trained Albert model with TensorRT.
The script is listed as follow:
```python
import torch_blade
import torch_blade.tensorrt
from torch_blade.testing.common_utils import assert_almost_equal
from transformers import AlbertConfig, AlbertModel, AutoTokenizer

def load_albert_and_inputs():
    model_name = "albert-base-v2"
    model = AlbertModel.from_pretrained(model_name, torchscript=True).cuda()

    tokenizer = AutoTokenizer.from_pretrained(model_name)
    def plain_tokenizer(inputs_str, return_tensors="pt"):
        inputs = tokenizer(inputs_str, return_tensors=return_tensors, padding=True)
        inputs = dict(map(lambda x: (x[0], x[1].cuda()), inputs.items()))

    input_strs = [
     "We are very happy to show you the story.",
     "We hope you don't hate it."]
    inputs = plain_tokenizer(input_strs)
    inputs = (inputs['input_ids'].cuda(), inputs['attention_mask'].cuda(), inputs['token_type_ids'].cuda(),)

    return model.eval(), inputs

def do_optimization(model, inputs):
    torch_config = torch_blade.Config()
    torch_config.optimization_pipeline = torch_blade.tensorrt.backend_name()
    with torch_config:
           opt_model = torch_blade.optimize(model, True, inputs)

    return opt_model

model, inputs = load_albert_and_inputs()
opt_model = do_optimization(model, inputs)

outputs0 = model(*inputs)
outputs1 = opt_model(*inputs)
assert_almost_equal(outputs0, outputs1)
```

After debugging, we found the accuracy error occurred because we use inputs with `torch.int64` data types. But TensorRT has only limited support on INT64. We fix up the issue by casting all `torch.LongTensor` to `torch.IntTensor` on the inputs of the subgraph given to ONNX2TensorRT. Refer to https://github.com/onnx/onnx-tensorrt/blob/main/docs/operators.md.

This pull request only fixes the accuracy error since TensorRT 8.2.3.0. TensorRT with a lower version has other accuracy error problems, but we won't fix them now.

### CODE REVIEW

File: pytorch_blade/torch_blade/onnx_backends/backend_conversion.py [hunk 1]
```diff
@@ -37,7 +39,21 @@ def _deduplicate_onnx_graph_outputs(graph):
...
+    # Refer to https://github.com/onnx/onnx-tensorrt/blob/main/docs/operators.md.
+    # There is limited support for INT32, INT64, and DOUBLE types
+    cfg = Config.get_current_context_or_new()
+    need_fixup_backends = ["TensorRT"]
+    if cfg.optimization_pipeline not in need_fixup_backends:
+        return
+    # Convert INT64 to INT32.
+    for val in graph.inputs():
+        if val.type().scalarType() == "Long":
+            tools.cast_to_i32_tensor_type(val)
```
Comment (bddppq) [inline pytorch_blade/torch_blade/onnx_backends/backend_conversion.py:52]: Where does this `cast_to_i32_tensor_type` function come from? I searched our codebase but couldn't see its definition. I'm wondering how could we fix issue at build time. Does it mutate the graph?

File: pytorch_blade/src/compiler/tensorrt/tensorrt_engine_context.cpp [hunk 2]
```diff
@@ -343,6 +341,9 @@ torch::List<torch::Tensor> TRTContext::PreProcessInputs(
     // add to cuda_ctu_inputs, make the ctu_tensor has lifetime out of the loop
     cuda_ctu_inputs.push_back(ctu_tensor);
   }
+
+  TORCH_CHECK(CheckCurrentDevice(ctu_inputs));
```
Comment (bddppq) [inline pytorch_blade/src/compiler/tensorrt/tensorrt_engine_context.cpp:345]: I think you meant `cuda_ctu_tensor`. Why do we need to move the check from the beginning to the end?

File: pytorch_blade/torch_blade/pass_manager.py [hunk 3]
```diff
@@ -121,7 +121,7 @@ def _jit_pass_lower_to_onnx(graph):
     #
     # But something weird happened:
     # call to ConstantValueMap::ClearMaps() in C++ will segfault
-    if utils.torch_version_number() >= utils.parse_version("1.11.0"):
+    if utils.torch_version_number() >= utils.parse_version("1.9.0"):
```
Comment (bddppq) [inline pytorch_blade/torch_blade/pass_manager.py:124]: What's special about this version?

File: pytorch_blade/torch_blade/onnx_backends/backend_conversion.py (hunk 1 above)
Comment (fortianyou) [reply pytorch_blade/torch_blade/onnx_backends/backend_conversion.py]: I had missed adding it.

File: pytorch_blade/src/compiler/tensorrt/tensorrt_engine_context.cpp (hunk 2 above)
Comment (fortianyou) [reply pytorch_blade/src/compiler/tensorrt/tensorrt_engine_context.cpp:345]: Thanks to remind me.

File: pytorch_blade/torch_blade/pass_manager.py (hunk 3 above)
Comment (fortianyou) [reply pytorch_blade/torch_blade/pass_manager.py:124]: The API of some ONNX passes has been changed since 1.9.0 but not 1.11.0.

File: pytorch_blade/src/compiler/jit/tool_funcs.cpp [hunk 4]
```diff
@@ -138,6 +138,16 @@ torch::TypePtr fromNumberType(torch::TypePtr typ) {
   return nullptr;
 }
 
+bool cast_to_i32_tensor_type(torch::jit::Value& value) {
```
Comment (bddppq) [inline pytorch_blade/src/compiler/jit/tool_funcs.cpp:141]: Looks to me here just simply change the input's scalar type at build time? But I think at runtime user will still pass long values to the trt engine, wouldn't that cause type mismatch?

Comment (fortianyou) [reply pytorch_blade/src/compiler/jit/tool_funcs.cpp:141]: In [`TRTContext::PreProcessInputs`](https://github.com/fortianyou/BladeDISC/blob/features/fix_trt/pytorch_blade/src/compiler/tensorrt/tensorrt_engine_context.cpp#L334)  will convert the input type to the corresponding graph input type at runtime. And the graph input type is recorded at build time.

Comment (bddppq) [reply pytorch_blade/src/compiler/jit/tool_funcs.cpp:141]: I see

File: pytorch_blade/torch_blade/onnx_backends/backend_conversion.py [hunk 5]
```diff
@@ -37,7 +39,20 @@ def _deduplicate_onnx_graph_outputs(graph):
         visited.add(v)
 
 
-def _build_onnx_engine(subgraph, engine_build_func, q_info=None, dynamic_settings=None):
+def _try_cast_graph_integer_inputs_to_i32(graph):
+    # Refer to https://github.com/onnx/onnx-tensorrt/blob/main/docs/operators.md.
+    # There is limited support for INT32, INT64, and DOUBLE types.
+    # Convert INT64 to INT32.
+    for val in graph.inputs():
+        if val.type().scalarType() == "Long":
```
Comment (bddppq) [inline pytorch_blade/torch_blade/onnx_backends/backend_conversion.py:47]: Maybe we should also handle double -> float conversion? (or at least add a todo?)

**Review by bddppq — APPROVED** [review]
LGTM

**Review by Orion34-lanbo — APPROVED** [review]
LGTM
